### PR TITLE
fix: circual dependancies between js-cloud-server and shared libs

### PR DESCRIPTION
**Pull request by jonathannorris**
- Moved `DevCycleOptions` to `@devcycle/types`

### CODE REVIEW

**Comment by jonathannorris** [comment]
Current dependencies on/for this PR:
* main
  * **PR #594** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/594" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/DevCycleHQ/js-sdks/594?utm_source=stack-comment).

File: lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { DVCLogger, DVCDefaultLogLevel } from '../../../logger'
+import { DVCReporter } from '../../../reporter'
+
+/**
+ * Options to control the setup of the DevCycle NodeJS Server SDK.
+ */
+export interface DevCycleOptions {
```
Comment (ajwootto) [inline lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts:7]: DevCycleServerSDKOptions?

Comment (jonathannorris) [reply lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts:7]: I think this is re-exported as a public interface for the SDKs so would likely need to be renamed there...

File: lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+     * Metrics reporter to capture data about event processing
+     */
+    reporter?: DVCReporter
+
+    /**
+     * Set log level of the default logger
+     */
+    logLevel?: DVCDefaultLogLevel
+
+    /**
```
Comment (ajwootto) [inline lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts:23]: does it really make sense to share this type around to all the different server packages. It feels like many of these options are irrelevant to the regular js server SDK, and shouldn't show up in the autocomplete of someone using it

Comment (ajwootto) [reply lib/shared/types/src/types/apis/sdk/serverSDKTypes.ts:23]: what about things like enableCloudBucketing and all the stuff related to event flushing and config refetching?

**Pull request merged**